### PR TITLE
Add submitOnEnter prop to TextField.TextArea

### PR DESCRIPTION
**Pull request by mathisobadia**
This PR adds a submitOnEnter prop to TextField.TextArea component.

This is useful in contexts where you want a textarea that can be auto resized but also want the input to submit a form when the enter key is pressed which is not the common behavior of a native textarea component contrary to a regular input.

The main use case is an input for a chat interface that can have multiple lines but should submit on enter (but not if shift key is pressed at the same time)

If submitOnEnter is set to true, then property `aria-multiline` is set to false to respect accessibility and indicate that enter will not insert line break but will submit

### CODE REVIEW

File: packages/core/src/text-field/text-field-text-area.tsx [hunk 1]
```diff
@@ -47,8 +48,23 @@ export function TextFieldTextArea(
     })
   );
 
+  const onKeyPressed = (event: KeyboardEvent) => {
+    if (ref && local.submitOnEnter && event.key === "Enter" && !event.shiftKey) {
```
Comment (fabien-ml) [inline packages/core/src/text-field/text-field-text-area.tsx:52]: I assume `shift` + `Enter` will create a line break, but how we allow line break on touch screen ?

Comment (mathisobadia) [reply packages/core/src/text-field/text-field-text-area.tsx:52]: Oh yes you are right, so looking at many chat implementations from chat gpt to intercom and facebook messenger it looks like they all do the same thing where the enter key submits but not on mobile where it just inserts a line break. So to make it similar I think we need to detect if the user is on mobile. I see there is already code to detect if user is ios and android, but I don't know if that is sufficient

Comment (fabien-ml) [reply packages/core/src/text-field/text-field-text-area.tsx:52]: I think a better will be to check if the user is using a virtual keyboard, but don't know if this API even exist 😅

Comment (mathisobadia) [reply packages/core/src/text-field/text-field-text-area.tsx:52]: One thing to note still on your initial question is that you can insert a line break on mobile in the same way as on desktop, if you click on the shift key of the virtual keyboard

Comment (mathisobadia) [reply packages/core/src/text-field/text-field-text-area.tsx:52]: I just checked the chat.openai.com implementation and they are doing a media query to know wether to submit on enter. If your screen is larger than 769px it submits, and it doesn't submit if it's smaller than that. Obviously we don't want to include that kind of logic into kobalte so this is probably left to consumer's code. I think we can leave the code as is as it has the same behavior on mobile and desktop and users can still insert line breaks on mobile by using the shift key in the virtual keyboard. And any other logic to disable submitOnEnter on mobile can be created by the consumer's code if they want to implement it

**Pull request merged**